src/cli/commands/force/lightning/local/setup.ts

```diff
@@ -67,16 +77,37 @@ export class Setup extends SfdxCommand implements HasRequirements {
     private _commandRequirements: CommandRequirements = {};
     public get commandRequirements(): CommandRequirements {
         if (Object.keys(this._commandRequirements).length === 0) {
-            const requirements = CommandLineUtils.platformFlagIsAndroid(
-                this.flags.platform
-            )
-                ? new AndroidEnvironmentRequirements(
-                      this.logger,
-                      this.flags.apilevel
-                  )
-                : new IOSEnvironmentRequirements(this.logger);
-
-            this._commandRequirements.setup = requirements;
+            if (CommandLineUtils.platformFlagIsAndroid(this.flags.platform)) {
```

Comment:
hmm... not sure if I like this approach. `setup.ts` was supposed to be the place where we look at the selected platform then call upon the platform specific environment requirements to be checked (i.e `AndroidEnvironmentRequirements` or `IOSEnvironmentRequirements`) and abstract away the list of requirements per platform.

Maybe I'm missing something but based on the changes in this file, I don't see why we need to move away from that design.

Reply:
The issue was that the current model is a single task with all 6 subtasks.  To add optional install tasks for all tasks (if we want to eventually resolve all tasks with installs), it would mean we would have 1 task with 12 subtasks.  Of which 6 would be skipped. You'd have to look carefully at the description to figure out which task was which.  Instead, I thought it made sense to make each "component" as separate task with two subtasks.  One to check and one to install.  So you can see here, you can clearly see each task.  And the success and status of each subtask.

![Screen Shot 2022-03-03 at 3 40 32 PM](https://user-images.githubusercontent.com/8267514/157118053-51d6d197-49cc-4da2-a4f1-74c528ac1166.png)
.

Reply:
The original list showed "Requirements" with a long list of requirements.  Not really breaking things into the sub-components.  For example.  These are just the component checks, without the install.  WIth installs, there would be another 6 under the main "Requirements" heading.
![Screen Shot 2022-03-03 at 3 51 46 PM](https://user-images.githubusercontent.com/8267514/157118326-85b0c708-a0ca-43aa-b721-41e59b077eac.png)

Reply:
But you could still move the lines here for setting up r1-r5 to `AndroidEnvironmentRequirements` and still achieve the same thing, right? That way you don't need to modify this file at all.

Reply:
ok, I'd have to modify the IOSEnvironmentRequirements file to return a single command key, but yes, I think that would make sense to pull this logic into Android and have setup agnostic of the tasks for each platform..